### ISSUE TITLE
Specify `module` to `build/index.esm.js` in `package.json` to support ESM import

### ISSUE THREAD

**Issue by Tanimodori** [ngVue/ngVue#143]
Currently the `main` script is set to UMD module `build/index.js`. But in some case if we need to use `vite` `transformMixedEsModules` function (https://github.com/vueComponent/ant-design-vue/issues/5717) like:

```ts
export default {
  build: {
    commonjsOptions: {
      transformMixedEsModules: true,
    },
  },
}
```

These lines in `build/index.js` will cause problems: (Claiming `Vue__default["default"].component is not a function`)

```ts
if (isCompositionApi(component)) {
  component = Vue__default["default"].component(component.name || 'UnnamedComponent', component);
}
//            ^ Vue__default["default"].component is not a function
```

Current workaround is manually resolving `ngVue` to `build/index.esm.js`:

```ts
export default {
  resolve: {
    alias: {
      ngVue: resolve(__dirname, './node_modules/ngVue/build/index.esm.js'),
    },
  },
}
```

I'd appreciate it that `module` is set to `build/index.esm.js` in `package.json`

**Comment by jaredmcateer** [comment]
Sure I'll try and take a look at this later today or tomorrow.

**Comment by jaredmcateer** [comment]
Published v2.1.1, let me know if you are still experiencing issues.
